### ISSUE TITLE
Support for possible future upgrades.

### ISSUE THREAD

**Issue by GuNanOvO** [VolmitSoftware/Iris#678]
Support for possible future upgrades.
After long-term active work by developers, the current iris has gradually approached a stable development. At this time, should we improve the upgrade support for the iris world?
I started using iris to create worlds on my server from version 1.6.7. Iris is a good enough world generation engine to bring players a beautiful world.
But now, I am considering how to upgrade from iris1.6.7 to the latest iris version, and the terrain data package should be upgraded along with it.
In the upcoming version 1.18 of Minecraft, how should my server follow up with the version? If I do not upgrade iris now, is it possible that iris 1.6.7 will continue to run well on the Minecraft 1.18 server?
When the overworld terrain pack adds new content and becomes better, how to upgrade from the old terrain pack to the new terrain pack?
There is no doubt that iris will get better and better in the future, with better performance and more content, but if there is no upgrade support for the old iris world, this will cause trouble for users of the old version of iris.
I checked the iris document, searched the discord-related chat records, and did not find enough support for this issue. I need help, and I think this is a reflection on the future development of iris, so I am here to ask questions, hoping to get answers from the lovely developers of iris.
My English is not good, I use Google Translate, hope to forgive me.

**Comment by ghost** [comment]
@GuNanOvO read this https://github.com/VolmitSoftware/Iris/issues/359
TLDR: There is no proper solution for upgrading the plugin or the pack. It will always corrupt the old chunks with bad transitions etc. Therefore its recommended to create a new world

**Comment by GuNanOvO** [comment]
Oops, the iris 1.6.7 version, which has been working well, went wrong today. Players discovered an error in the generation of the iris world today. The red glazed terracotta was generated at the bottom of the world, and no terrain was generated in a large area. Since a few days ago, when there were players in the iris world, iris kept throwing warnings in the background. I didn't notice the generation error, so I didn't care about it. But what I was worried about still happened.
I am very troubled now, if I regenerate the iris world, the players will definitely be angry. If you upgrade to a new version of iris and generate a new iris world, will the new iris world be upgraded to Minecraft 1.18 when Minecraft 1.18 is released in the future?

**Comment by ghost** [comment]
You mean upgradeable without any unmachting chunk borders or are you trying to ask if iris will support 1.18 in the future?

**Comment by GuNanOvO** [comment]
> You mean upgradeable without any unmachting chunk borders or are you trying to ask if iris will support 1.18 in the future?

I'm sorry to confuse you. But what I want to ask is: now use the latest iris to generate an iris world. Can this world continue to be used on minecraft 1.18 server in the future?

**Comment by GuNanOvO** [comment]
I didn't intend to upgrade iris, but iris 1.6.7, which has been running well, made an error and the iris world was damaged.
![2021-10-23_10 15 38](https://user-images.githubusercontent.com/63027311/138538763-3e59a025-0b79-47a6-9cb8-fcd662335f63.png)
![2021-10-23_10 15 57](https://user-images.githubusercontent.com/63027311/138538770-9d95ce29-d2f6-49ab-9406-3a23770d42e9.png)

**Comment by ghost** [comment]
To quote the dev of iris:
„ Generally its not possible to do something like that without ending up with hard borders / having to edit millions of chunks possibly“
„ By changing the biome map, everything changes. Things that were a desert are now a forest.“
„ I could blend it by using an interpolator, but... ignoring the intense cpu from that, it would still look terrible, because it would still look like a wall, just smoother, and the biomes would still be hard-cut. Just update when your ready to make a new map, its not a big issue 😛„
I therefore close this
